sdk/keyvault/tests.yml

```diff
@@ -6,6 +6,7 @@ extends:
     ServiceDirectory: keyvault
     MaxParallel: 5
     TimeoutInMinutes: 180
+    Clouds: 'Public,Preview,UsGov,China,Canary'
```

Comment:
Can you update the `CloudConfig` value to the snippet below?

```
    CloudConfig:
      Public:
        SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
      Canary:
        SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
        Location: 'eastus2euap'
        # Managed HSM test resources are expensive and provisioning has not been reliable.
        # Given test coverage of non-canary regions we probably don't need to test in canary.
        MatrixFilters:
         - ArmTemplateParameters=^(?!.*enableHsm.*true)
      UsGov:
        SubscriptionConfiguration: $(sub-config-gov-test-resources)
      China:
        SubscriptionConfiguration: $(sub-config-cn-test-resources)
```

The reason we are overriding the whole `CloudConfig` is because it's the only way to pass in the `MatrixFilters` field for `Canary` since there are a few things we need to skip there.

@heaths do you think it's okay to enable the HSM tests for usgov+china too, or should we also exclude them?

Reply:
According to https://azure.microsoft.com/global-infrastructure/services/?products=key-vault it should be available, so we could do that.

Reply:
@benbp  l have updated the CloudConfig value.

Reply:
The resource type `Microsoft.KeyVault/managedHSMs` was invalid in the namespace `Microsoft.KeyVault` for UsGov and China clouds. Shall we exclude them?

Reply:
@zedy-wj yes you can follow the same pattern we're using to exclude them from canary: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/tests.yml#L17-L18 (copy/paste the matrix filters for usgov and china cloud config entries).

Reply:
Is it unavailable or does it need a different API version @heaths:

```
          "message": "The resource type could not be found in the namespace 'Microsoft.KeyVault' for api version '2020-04-01-preview'."
```